packages/crdt/src/crdt/merkle.ts

```diff
@@ -19,12 +19,18 @@ export type TrieNode = {
   hash?: number;
 };
 
+type NumberTrieNodeKey = keyof Omit<TrieNode, 'hash'>;
+
 export function emptyTrie(): TrieNode {
   return { hash: 0 };
 }
 
-export function getKeys(trie: TrieNode): ('0' | '1' | '2')[] {
-  return Object.keys(trie).filter(x => x !== 'hash') as ('0' | '1' | '2')[];
+function isNumberTrieNodeKey(input: string): input is NumberTrieNodeKey {
+  return ['0', '1', '2'].includes(input);
+}
+
+export function getKeys(trie: TrieNode): NumberTrieNodeKey[] {
+  return Object.keys(trie).filter(isNumberTrieNodeKey);
 }
```

Comment:
Tiny improvement to the `getKeys` function. Previously we were manually casting the return type to `as YYY...`. But now we have a type-guard (`isNumberTrieNodeKey`) that allows us to drop this manual type coercion.

Functionally the code is the same.